hazelcast/src/main/java/com/hazelcast/query/impl/TypeConverters.java

```diff
@@ -173,8 +184,23 @@ Comparable convertInternal(Comparable value) {
             if (value instanceof BigInteger) {
                 return new BigDecimal((BigInteger) value);
             }
+            if (isNotFloatingPointNumber(value)) {
+                Number number = (Number) value;
+                return BigDecimal.valueOf(number.longValue());
+            }
+            if (value instanceof Double) {
+                Number number = (Number) value;
+                return BigDecimal.valueOf(number.doubleValue());
+            }
```

Comment:
Just a pickle here, the BigDecimal.valueOf uses the Double.toString(), which makes it slow too. Here's the snippet.
```
public static BigDecimal valueOf(double val) {
        // Reminder: a zero double returns '0.0', so we cannot fastpath
        // to use the constant ZERO.  This might be important enough to
        // justify a factory approach, a cache, or a few private
        // constants, later.
        return new BigDecimal(Double.toString(val));
    }
```
It would be better tom use the `new BigDecimal(double)`.
WDYT?